### PR TITLE
test(JestMigration): ReferenceDotSpec.js -> ReferenceDot.spec.tsx

### PR DESCRIPTION
**Pull request by proke03**
## Description
Migrate ReferenceDotSpec.js -> ReferenceDot.spec.tsx

## Related Issue
#3190

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: test-jest/cartesian/ReferenceDot.spec.tsx [hunk 1]
```diff
@@ -74,17 +73,17 @@ describe('<ReferenceDot />', () => {
...
-    expect(wrapper.find('.recharts-reference-dot-dot').length).to.equal(1);
-    expect(wrapper.find('.recharts-label').length).to.equal(1);
+    expect(container.querySelectorAll('.recharts-reference-dot-dot')).toHaveLength(1);
+    expect(container.querySelectorAll('.recharts-label')).toHaveLength(1);
   });
 
-  it('Render custom lable when label is set to be a react element', () => {
-    const Label = ({ text, ...props }) => (
+  test('Render custom lable when label is set to be a react element', () => {
+    const Label = ({ text, ...props }: { text: any }) => (
```
Comment (nikolasrieble) [inline test-jest/cartesian/ReferenceDot.spec.tsx:81]: Please do not use any if possible.

File: test-jest/cartesian/ReferenceDot.spec.tsx [hunk 2]
```diff
@@ -123,16 +122,17 @@ describe('<ReferenceDot />', () => {
...
       </BarChart>,
     );
-    expect(wrapper.find('.customized-label').text()).to.equal('Custom Text');
+    expect(screen.findByText('Custom Text')).toBeTruthy();
   });
 
-  it("Don't render any label when label is a plain object", () => {
-    const renderLabel = ({ text, ...props }) => (
+  test("Don't render any label when label is a plain object", () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nikolasrieble) [inline test-jest/cartesian/ReferenceDot.spec.tsx:129]: Is this needed? Can we write the code such that this is not needed?

File: test-jest/cartesian/ReferenceDot.spec.tsx (hunk 1 above)
Comment (proke03) [reply test-jest/cartesian/ReferenceDot.spec.tsx:81]: I think it's ok in test code.

Comment (ckifer) [reply test-jest/cartesian/ReferenceDot.spec.tsx:81]: `any` is still used here - `text` should be a string at the very least - fine for the sake of the migration but lets try to be strict with our typing

**Review by ckifer — APPROVED** [review]

**Pull request merged**